x-pack/plugins/index_management/__jest__/client_integration/helpers/http_requests.ts

```diff
@@ -17,139 +18,105 @@ export interface ResponseError {
-      JSON.stringify(response),
-    ]);
-  };
-
-  const setDeleteTemplateResponse = (response: HttpResponse = []) => {
-    server.respondWith('POST', `${API_BASE_PATH}/delete_index_templates`, [
-      200,
-      { 'Content-Type': 'application/json' },
-      JSON.stringify(response),
-    ]);
-  };
-
-  const setLoadTemplateResponse = (response?: HttpResponse, error?: any) => {
-    const status = error ? error.status || 400 : 200;
-    const body = error ? error.body : response;
-
-    server.respondWith('GET', `${API_BASE_PATH}/index_templates/:id`, [
-      status,
-      { 'Content-Type': 'application/json' },
-      JSON.stringify(body),
-    ]);
-  };
-
-  const setCreateTemplateResponse = (response?: HttpResponse, error?: any) => {
-    const status = error ? error.body.status || 400 : 200;
-    const body = error ? JSON.stringify(error.body) : JSON.stringify(response);
-
-    server.respondWith('POST', `${API_BASE_PATH}/index_templates`, [
-      status,
-      { 'Content-Type': 'application/json' },
-      body,
-    ]);
-  };
-
-  const setUpdateTemplateResponse = (response?: HttpResponse, error?: any) => {
-    const status = error ? error.status || 400 : 200;
-    const body = error ? JSON.stringify(error.body) : JSON.stringify(response);
-
-    server.respondWith('PUT', `${API_BASE_PATH}/index_templates/:name`, [
-      status,
-      { 'Content-Type': 'application/json' },
-      body,
-    ]);
-  };
-
-  const setUpdateIndexSettingsResponse = (response?: HttpResponse, error?: ResponseError) => {
-    const status = error ? error.statusCode || 400 : 200;
-    const body = error ?? response;
-
-    server.respondWith('PUT', `${API_BASE_PATH}/settings/:name`, [
-      status,
-      { 'Content-Type': 'application/json' },
-      JSON.stringify(body),
-    ]);
-  };
-
-  const setSimulateTemplateResponse = (response?: HttpResponse, error?: any) => {
-    const status = error ? error.status || 400 : 200;
-    const body = error ? JSON.stringify(error.body) : JSON.stringify(response);
-
-    server.respondWith('POST', `${API_BASE_PATH}/index_templates/simulate`, [
-      status,
-      { 'Content-Type': 'application/json' },
-      body,
-    ]);
-  };
-
-  const setLoadComponentTemplatesResponse = (response?: HttpResponse, error?: any) => {
-    const status = error ? error.status || 400 : 200;
-    const body = error ? error.body : response;
-
-    server.respondWith('GET', `${API_BASE_PATH}/component_templates`, [
-      status,
-      { 'Content-Type': 'application/json' },
-      JSON.stringify(body),
-    ]);
-  };
-
-  const setLoadNodesPluginsResponse = (response?: HttpResponse, error?: any) => {
-    const status = error ? error.status || 400 : 200;
-    const body = error ? error.body : response;
-
-    server.respondWith('GET', `${API_BASE_PATH}/nodes/plugins`, [
-      status,
-      { 'Content-Type': 'application/json' },
-      JSON.stringify(body),
-    ]);
-  };
+const registerHttpRequestMockHelpers = (
```

Comment:
Starting here to L56 seems to be the same code across the test refactoring PRs. Does it make sense to try to centralize it somewhere so it can be reused?

Reply:
If you agree, I think this can be done as a separate PR

Reply:
I've suggested also to put this somewhere in `kbn-test` (see [this thread](https://github.com/elastic/kibana/pull/127122#discussion_r824585861)), but we ended up agreeing on keeping separate implementations per app for now. I'll create a new issue on our side to update this later on.